cmd/sonobuoy/app/rbac.go

```diff
@@ -0,0 +1,94 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// RBACMode determines whether to enable or disable RBAC for a Sonobuoy run
+type RBACMode string
+
+var (
+	RBACErrorNoClient = errors.New("can't use nil client with \"detect\" RBAC mode")
+)
+
+const (
+	// DisableRBACMode means rbac is always disable
+	DisableRBACMode RBACMode = "disable"
+	// EnabledRBACMode means rbac is always enabled
+	EnabledRBACMode RBACMode = "enabled"
+	// DetectRBACMode means "query the server to see if RBAC is enabled"
+	DetectRBACMode RBACMode = "detect"
+)
+
+var rbacModeMap = map[string]RBACMode{
+	string(DisableRBACMode): DisableRBACMode,
+	string(EnabledRBACMode): EnabledRBACMode,
+	string(DetectRBACMode):  DetectRBACMode,
+}
+
+// String needed for pflag.Value.
+func (r *RBACMode) String() string { return string(*r) }
+
+// Type needed for pflag.Value.
+func (r *RBACMode) Type() string { return "RBACMode" }
+
+// Set the RBACMode to the given string, or error if it's not a known RBAC mode.
+func (r *RBACMode) Set(str string) error {
+	mode, ok := rbacModeMap[str]
+	if !ok {
+		return fmt.Errorf("unknown RBAC mode %s", str)
+	}
+	*r = mode
+	return nil
+}
+
+const apiRootPath = "/apis"
+
+var supportedRBACGroupVersion = rbacv1.SchemeGroupVersion.String()
+
+// Get retrieves whether to enable or disable rbac. If the mode is disable or
+// enabled, the client is unused. If the mode is "detect", the client will be
+// used to query the server's API groups and detect whether an RBAC api group exists.
+func (r *RBACMode) Get(client *kubernetes.Clientset) (bool, error) {
```

Comment:
Get is the convention for all the other `pflag.Value`s. I think we should only change it if we decide we don't like that convention.

Reply:
I don't like that convention, I think it's overly generic. It could be more clear perhaps if the type were called RBACModeEnabled, then we'd have .Get on that and that makes more sense to me, but I guess I don't really care that much. If you dig this pattern, we can keep it.